contracts/modules/staking/StakingPool.sol

```diff
@@ -690,27 +711,64 @@ contract StakingPool is IStakingPool, Multicall {
       rewardPerSecond += _rewardPerSecond.toUint96();
 
       uint rewardsToMint = _rewardPerSecond * rewardStreamPeriod;
+
       tokenController.mintStakingPoolNXMRewards(rewardsToMint, poolId);
     }
 
-    // remove previous rewards
-    if (previousPremium > 0) {
+    return (premium, allocationId);
+  }
 
-      uint prevRewards = previousPremium * request.previousRewardsRatio / REWARDS_DENOMINATOR;
-      uint prevExpirationBucket = Math.divCeil(request.previousExpiration, BUCKET_DURATION);
-      uint rewardStreamPeriod = prevExpirationBucket * BUCKET_DURATION - request.previousStart;
-      uint prevRewardsPerSecond = prevRewards / rewardStreamPeriod;
+  function getPremium(
+    uint amount,
+    uint previousAllocationAmountInNXMRepriced,
+    uint coverAllocationAmount,
+    uint initialCapacityUsed,
+    uint totalCapacity,
+    AllocationRequest calldata request
+  ) internal returns (uint totalPremium) {
 
-      // store
-      rewardPerSecondCut[prevExpirationBucket] -= prevRewardsPerSecond;
-      rewardPerSecond -= prevRewardsPerSecond.toUint96();
+    if (amount > previousAllocationAmountInNXMRepriced) {
+
+      // this handles both new allocations (allocationId == 0) and edits that increase the amount
```

Comment:
<img width="649" alt="Screenshot 2023-09-15 at 9 46 17 AM" src="https://github.com/NexusMutual/smart-contracts/assets/1534891/989d76f0-8d80-4105-a7f1-7c819f23dc41">
Merged the computation logic here again because of contract size